.github/workflows/docs.yml

```
@@ -0,0 +1,34 @@
+name: Build and Deploy Sphinx Docs
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.10'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r default-image/requirements.txt
+        pip install sphinx sphinx_rtd_theme
+
+    - name: Build Sphinx documentation
+      run: |
+        cd docs
+        make html
+
+    - name: Deploy to GitHub Pages
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./docs/build/html
```

I think this needs to be indented one level. Also swapped the action used to deploy; GitHub maintains their own action for publishing to GH pages-- might be better to rely on theirs, rather than a 3rd party?

```suggestion
    steps:
      - uses: actions/checkout@v2

      - name: Set up Python
        uses: actions/setup-python@v2
        with:
          python-version: '3.10'

      - name: Install dependencies
        run: |
          python -m pip install --upgrade pip
          pip install -r default-image/requirements.txt
          pip install sphinx sphinx_rtd_theme

      - name: Build Sphinx documentation
        run: |
          cd docs
          make html

      - name: Deploy to GitHub Pages
        uses: actions/deploy-pages@v4
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./docs/build/html
```